From patchwork Tue Sep 18 15:51:18 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Peng Hao <penghao122@sina.com.cn>
X-Patchwork-Id: 10604541
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 7E15815A6
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 18 Sep 2018 16:13:07 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 6EE2C286D5
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 18 Sep 2018 16:13:07 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 6376F28718; Tue, 18 Sep 2018 16:13:07 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id A396A286F6
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue, 18 Sep 2018 16:13:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1729618AbeIRVqU (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 18 Sep 2018 17:46:20 -0400
Received: from mail115-79.sinamail.sina.com.cn ([218.30.115.79]:48124 "HELO
        mail115-79.sinamail.sina.com.cn" rhost-flags-OK-OK-OK-OK)
        by vger.kernel.org with SMTP id S1729197AbeIRVqU (ORCPT
        <rfc822;kvm@vger.kernel.org>); Tue, 18 Sep 2018 17:46:20 -0400
X-Greylist: delayed 1294 seconds by postgrey-1.27 at vger.kernel.org;
 Tue, 18 Sep 2018 17:46:19 EDT
Received: from unknown (HELO
 localhost.localdomain.localdomain)([110.184.36.9])
        by sina.com with ESMTP
        id 5BA11EF800003E41; Tue, 18 Sep 2018 23:51:27 +0800 (CST)
X-Sender: penghao122@sina.com.cn
X-Auth-ID: penghao122@sina.com.cn
X-SMAIL-MID: 403331131988
From: Peng Hao <penghao122@sina.com.cn>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: kvm@vger.kernel.org, Peng Hao <peng.hao2@zte.com.cn>
Subject: [PATCH]  kvm-unit-tests: unlock before calling return
Date: Tue, 18 Sep 2018 11:51:18 -0400
Message-Id: <1537285878-27148-1-git-send-email-penghao122@sina.com.cn>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Peng Hao <peng.hao2@zte.com.cn>

Signed-off-by: Peng Hao <peng.hao2@zte.com.cn>
Reviewed-by: Krish Sadhukhan <krish.sadhukhan@oracle.com>
---
 lib/report.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/lib/report.c b/lib/report.c
index 5da27ab..ca9b4fd 100644
--- a/lib/report.c
+++ b/lib/report.c
@@ -66,8 +66,10 @@ void report_prefix_pop(void)
 
 	spin_lock(&lock);
 
-	if (!*prefixes)
+	if (!*prefixes) {
+		spin_unlock(&lock);
 		return;
+	}
 
 	for (p = prefixes, q = strstr(p, PREFIX_DELIMITER) + 2;
 			*q;
@@ -142,6 +144,7 @@ void report_info(const char *msg_fmt, ...)
 
 int report_summary(void)
 {
+	int ret;
 	spin_lock(&lock);
 
 	printf("SUMMARY: %d tests", tests);
@@ -153,14 +156,16 @@ int report_summary(void)
 		printf(", %d skipped", skipped);
 	printf("\n");
 
-	if (tests == skipped)
+	if (tests == skipped) {
+		spin_unlock(&lock);
 		/* Blame AUTOTOOLS for using 77 for skipped test and QEMU for
 		 * mangling error codes in a way that gets 77 if we ... */
 		return 77 >> 1;
+	}
 
-	return failures > 0 ? 1 : 0;
-
+	ret = failures > 0 ? 1 : 0;
 	spin_unlock(&lock);
+	return ret;
 }
 
 void report_abort(const char *msg_fmt, ...)
